### PR TITLE
Stream

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: builder/src/templates/worker.ts [hunk 1]
```diff
@@ -23,6 +21,7 @@ export default {
         customServer: false,
         dev: false,
         dir: "",
+        minimalMode: false,
```
Comment (petebacondarwin) [inline builder/src/templates/worker.ts:24]: why this?

File: builder/src/templates/worker.ts [hunk 2]
```diff
@@ -32,50 +31,97 @@ export default {
...
+
+  // Only allow statusCode to be modified if not sent
+  let { statusCode } = res;
+  Object.defineProperty(res, "statusCode", {
+    get: function () {
+      return statusCode;
+    },
+    set: function (val) {
+      if (this.finished || this.headersSent) {
+        return;
```
Comment (petebacondarwin) [inline builder/src/templates/worker.ts:105]: Not a throw?

**Review by petebacondarwin — APPROVED** [review]

File: builder/src/templates/worker.ts (hunk 1 above)
Comment (vicb) [reply builder/src/templates/worker.ts:24]: This is to make the default explicit

File: builder/src/templates/worker.ts (hunk 2 above)
Comment (vicb) [reply builder/src/templates/worker.ts:105]: Adding a console error for now

File: builder/src/build/build-worker.ts [hunk 3]
```diff
@@ -100,12 +100,18 @@ globalThis.fetch = (input, init) => {
...
+import { Readable } from 'node:stream';
 globalThis.fetch.__nextPatched = isPatchedAlready;
 fetch = globalThis.fetch;
 const CustomRequest = class extends globalThis.Request {
   constructor(input, init) {
     console.log("CustomRequest", input);
-    if (init) delete init.cache;
+    if (init) {
+      delete init.cache;
+      if (init.body?.__node_stream__ === true) {
```
Comment (mhart) [inline builder/src/build/build-worker.ts:111]: I think it should be fine to do without the custom property and just check `instance of Readable`:

```
    if (init) {
      init = {
        ...init,
        cache: undefined,
        body: init.body instanceof Readable ? Readable.toWeb(init.body) : init.body
      }
    }
```

(at least, these works in my tests so far)

**Pull request merged**